Fix errors in UT for NoiseExtractTest

tszumski opened the pull request:
AVX2 code assume that width is divisible by 64.
Added support for width divisible by 16 and 32

ttrigui commented:
> AVX2 code assume that width is divisible by 64.
> Added support for width divisible by 16 and 32

Can you use the new AVX2 code instead of c code in 
if (sb_origin_x + BLOCK_SIZE_64 > input_picture_ptr->width)
            {
                noise_extract_luma_weak_c(
                    input_picture_ptr,
                    denoised_picture_ptr,
                    noise_picture_ptr,
                    sb_origin_y,
                    sb_origin_x);
            }
whenever the remaining width is divisible by 16 or 32?

ttrigui commented:
@tszumski can you please lokk into my previous comment?

tszumski commented:
Hi, 
I changed the code as you requested.
The changes:
1. Added variable "idx" in AVX2 function to match C implementation (Basically variable "sb_origin_x" was unused in AVX2)
2. Change conditions before "noise_extract_*" functions are called

hassount commented:
@tszumski  can we get this rebased ?
@joelsole @mafonsoN @andrey-norkin  any issues with this ?

tszumski commented:
@hassount 
Rebase done

andrey-norkin commented on Source/Lib/Common/ASM_AVX2/EbNoiseExtractAVX2.c:
```diff
@@ -66,6 +66,48 @@ inline void luma_weak_filter_avx2_intrin(
+    uint8_t                       *ptr_noise
+)
+{
+    __m128i topFirstHalf, bottomFirstHalf,
```
We are trying to get the code style in the main codec part in accordance with the style guidelines (can be found in this repo). According to guidelines, the variables and function names should be snake_case, whereas type names are CamelCase. This may not be uniformly followed throughout the code, but the new PRs should adhere to this rule. The only exception is the unit testing part, which is written in C++ and have somewhat different style agreement.

andrey-norkin commented on Source/Lib/Common/ASM_AVX2/EbNoiseExtractAVX2.c:
```diff
@@ -148,6 +190,75 @@ inline void chroma_weak_luma_strong_filter_avx2_intrin(
+)
+{
+    __m128i filterFirstHalf, filterSecondHalf,
+        currNextFirstHalf, currNextSecondHalf,
```
Same here

andrey-norkin commented on Source/Lib/Common/ASM_AVX2/EbNoiseExtractAVX2.c:
```diff
@@ -226,6 +337,62 @@ inline void chroma_strong_avx2_intrin(
+    uint8_t                       *ptr_denoised
+)
+{
+    __m128i   currLeftMidFirstHalflo, currLeftMidFirstHalfhi, topLeftMidFirstHalflo, topLeftMidFirstHalfhi,
```
Code style
